src/OpenApi/src/Schemas/OpenApiJsonSchema.Helpers.cs

```diff
@@ -0,0 +1,284 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+
+internal sealed partial class OpenApiJsonSchema
+{
+    /// <summary>
+    /// Read a list from the given JSON reader instance.
+    /// </summary>
+    /// <typeparam name="T">The type of the elements that will populate the list.</typeparam>
+    /// <param name="reader">The <see cref="Utf8JsonReader"/> to consume the list from.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> instance.</param>
+    /// <returns>A list parsed from the JSON array.</returns>
+    public static List<T>? ReadList<T>(ref Utf8JsonReader reader, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.Null)
+        {
+            return default;
+        }
+
+        if (reader.TokenType == JsonTokenType.StartArray)
+        {
+            var typeInfo = options.GetTypeInfo(typeof(T));
+            var valueConverter = (JsonConverter<T>)typeInfo.Converter;
+            var values = new List<T>();
+            reader.Read();
+            while (reader.TokenType != JsonTokenType.EndArray)
```

Comment:
We don't have to handle unexpected token types, as in `OpenApiJsonSchema.Read`?

Reply:
We could be more defensive  here if we wanted but I was a bit lax here since the JsonObject we're consuming is from a friendly and reliable party (STJ).

Reply:
It couldn't include a comment?